### ISSUE TITLE
[Feature request]: search by email field

### ISSUE THREAD

**Issue by jvanz** [nylas/N1#1161]
I think a good feature is allow user filter by email fields (e.g. to, cc, ccm, has attachment, etc)

**Comment by mbilker** [comment]
Thanks for the request @jvanz. Are you referring to mail filtering or the search bar? Mail filtering has this capability. The search bar passes the request to your provider. So for Gmail, you can use the same terms you normally use for searching in the Gmail web application.

**Comment by emorikawa** [comment]
Yes @jvanz our search currently passes the query directly to your mail provider. If you're backed by gmail, any search term you can natively type into Gmail should work in N1 as well.

We definitely should expose this better, however, and offer it on other mail providers.

**Comment by jvanz** [comment]
Good to know.

You're right.  Currently, I use N1 with 3 accounts ( 2 are not Gmail) and use
this Gmail search style in all of them will be awesome! =)

> On Feb 1 2016, at 2:42 pm, Evan Morikawa &lt;notifications@github.com&gt;
wrote:  

>

> Yes [@jvanz](https://github.com/jvanz) our search currently passes the query
directly to your mail provider. If you're backed by gmail, any search term you
can natively type into Gmail should work in N1 as well.

>

> We definitely should expose this better, however, and offer it on other mail
providers.

>

> —  
Reply to this email directly or [view it on GitHub](https://github.com/nylas/N
1/issues/1161#issuecomment-178061549).![](https://github.com/notifications/bea
con/ABcdLnYTuEUxuxg9i6JucY_ELKWlzbWiks5pf4JrgaJpZM4HPt74.gif)

**Comment by tbraeutigam** [comment]
Would it be possible to expose the same filters as the N1 API?
https://www.nylas.com/docs/platform/#filtering

ie, I can enter the following in the search and it'll do the same as filtering on the API:
`unread:false starred:true from:xyz@zyx.com filename:important.pdf`

**Comment by bengotow** [comment]
We've had another request for this from the support channel: https://nylas.zendesk.com/agent/tickets/7156
